src/models/timeaxisheaderview.cpp

```diff
@@ -0,0 +1,257 @@
+#include "timeaxisheaderview.h"
+
+#include <QPainter>
+#include <QtMath>
+#include <QDebug>
+
+#include "filterandzoomstack.h"
+
+TimeAxisHeaderView::TimeAxisHeaderView(FilterAndZoomStack *filterAndZoomStack, Qt::Orientation orientation, QWidget *parent)
+    : QHeaderView(orientation, parent)
+    , m_filterAndZoomStack(filterAndZoomStack)
+{
+    setMinimumHeight(40);
+
+    connect(filterAndZoomStack, &FilterAndZoomStack::filterChanged, this, &TimeAxisHeaderView::onFilterChanged);
+    connect(filterAndZoomStack, &FilterAndZoomStack::zoomChanged, this, &TimeAxisHeaderView::onZoomChanged);
+}
+
+void TimeAxisHeaderView::setTimeRange(const Data::TimeRange &timeRange)
+{
+    m_timeRange = timeRange;
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::onFilterChanged(const Data::FilterAction &filterAction)
+{
+    Q_UNUSED(filterAction)
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::onZoomChanged(const Data::ZoomAction &zoomAction)
+{
+    Q_UNUSED(zoomAction)
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::paintSection(QPainter *painter, const QRect &rect, int logicalIndex) const
+{
+    if (painter == nullptr) return;
+    if (logicalIndex != 1) return; // TODO: role based?
```

Comment:
I don't quite understand that comment, what has the role to do with the index?

Reply:
The timeline is only drawn above the events, this has been made clear in the code